lib/airrecord/faraday_rate_limiter.rb

```diff
@@ -0,0 +1,60 @@
+require 'thread'
+
+module Airrecord
+  class FaradayRateLimiter < Faraday::Middleware
+    class << self
+      attr_accessor :requests
+    end
+
+    def initialize(app, requests_per_second: nil, sleeper: nil)
+      super(app)
+      @rps = requests_per_second
+      @sleeper = sleeper || ->(seconds) { sleep(seconds) }
+      @mutex = Mutex.new
+      clear
+    end
+
+    def call(env)
+      @mutex.synchronize do
+        wait if too_many_requests_in_last_second?
+        @app.call(env).on_complete do |_response_env|
+          requests << Process.clock_gettime(Process::CLOCK_MONOTONIC)
+          requests.shift if requests.size > @rps
+        end
+      end
+    end
+
+    def clear
+      self.class.requests = []
```

Comment:
It seems like `#clear` is only used for teardown in the tests. Do we need it?

Reply:
Yeah, I could probably just reach into the class instead of extending the API 👍